### DISCUSSION TITLE
Automatic podcast downloads

### DISCUSSION THREAD

**Discussion by F1orian** [LeoKlaus/plappa#217] [Feature Requests]
As a user, I would like new episodes of podcasts to be automatically downloaded.

The number of episodes to download automatically should be configurable per podcast.

**Comment by LeoKlaus** [comment]
This is something I planned to add for a long time, but haven't gotten around to.

iOS heavily restricts background tasks and while it's technically possible to schedule a periodic refresh and download of episodes, the interval and duration of those tasks are at the discretion of the OS. This means that after an episode releases, it could potentially take hours or days until it is automatically downloaded.


The easiest solution is probably to implement #216 and have plappa start the download when it receives a push notification. ABS already does allow users to configure a notification server, so that part is easily solved.
The primary issue with this is that push notifications to a device can only be sent through Apples servers, which require a certificate for authentication that I obviously can't make available to users.

Luckily, this is an issue I have already solved for Paperparrot, which uses a relay server to handle push notifications, so this should be fairly easy to adapt for plappa.


I can't give any estimates as my main focus right now is the Apple Watch app, but I do want to add this _soon_.

**Reply by nupulu** [reply]
ShelfPlayer has somehow solved this. Take a look?

**Comment by GuyWithAQuestn** [comment]
Just adding a voice that if you have time for feature requests and wondering which one to work on..... this one, please!
Thank you!
